lib/widgets.php

```diff
@@ -284,3 +284,37 @@ function gutenberg_load_widget_preview_if_requested() {
 }
 add_filter( 'load-appearance_page_gutenberg-widgets', 'gutenberg_load_widget_preview_if_requested' );
 
+/**
+ * Sets show_instance_in_rest to true on all of the core WP_Widget subclasses.
+ * When merge dto Core, this property should be added to WP_Widget and set to
+ * true on each WP_Widget subclass.
+ */
+function gutenberg_set_show_instance_in_rest_on_core_widgets() {
+	global $wp_widget_factory;
+
+	$core_widgets = array(
+		'WP_Widget_Block',
+		'WP_Widget_Pages',
+		'WP_Widget_Calendar',
+		'WP_Widget_Archives',
+		'WP_Widget_Media_Audio',
+		'WP_Widget_Media_Image',
+		'WP_Widget_Media_Gallery',
+		'WP_Widget_Media_Video',
+		'WP_Widget_Meta',
+		'WP_Widget_Search',
+		'WP_Widget_Text',
+		'WP_Widget_Categories',
+		'WP_Widget_Recent_Posts',
+		'WP_Widget_Recent_Comments',
+		'WP_Widget_RSS',
+		'WP_Widget_Tag_Cloud',
+		'WP_Nav_Menu_Widget',
+		'WP_Widget_Custom_HTML',
+	);
+
+	foreach ( $core_widgets as $widget ) {
+		$wp_widget_factory->widgets[ $widget ]->show_instance_in_rest = true;
```

Comment:
Why is this param needed again? What is the plan for this when it goes into core.

Reply:
It's how the controller decides if the raw instance data is exposed.

Reply:
When in Core this would be added as a property to `WP_Widget` which defaults to `false`.

```php
class WP_Widget {
	...
	public $show_instance_in_rest = false;
	...
}
```

Widgets that wish to expose their raw instance over the REST API can set this property to `true` in their `WP_Widget` subclass. By doing so, the widget is declaring that only JSON-serializable values are stored into the instance settings. All Core widgets would set this property to `true`.

```php
class WP_Widget_Search extends WP_Widget {
	...
	public function __construct() {
		...
		$this→show_instance_in_rest = true;
		...
	}
	...
}
```

Reply:
Right, I get it, it is an opt-in for widgets because widget data might be sensitive, like an api key or something. 

I think personally that is a overkill, as the api is not public anyway, it does a capability check  meaning that data is not public. What is the point of hiding this data if you already have done a capability check? @noisysocks

Reply:
> By doing so, the widget is declaring that only JSON-serializable values are stored into the instance settings. 

I think this bit is important. One other reason, apart from sensitive data, to hide the raw data is if it is not JSON serializable, so you want to expose only encoded data if the items in your instance are not serializable.

Reply:
If a user has compabilities to see widgets the in admin terminal, when is there a time that couldn't see this data by just looking at the widget screen?

Reply:
This doesn't really have anything to do with permissions. It is for JSON serializability purposes. The base64 encoded `serialize` form of the instance is exposed regardless of `show_instance_in_rest`.

https://github.com/WordPress/gutenberg/blob/8fd16234839f20892952f103eb51ad17c6ca467a/lib/class-wp-rest-widgets-controller.php#L550